lib/job/index.js

```diff
@@ -33,6 +34,9 @@ class Job {
     this.agenda = args.agenda;
     delete args.agenda;
 
+    this.mongoOptions = args.mongoOptions || {};
+    delete args.mongoOptions;
```

Comment:
Why delete ?

Reply:
For sure!
Probably I follow the delete from line 35 without thinking about the consequences.

Reply:
The object should be cloned before mutating.